org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/wizards/AbapGitWizardPull.java

```diff
@@ -148,63 +156,73 @@ public void addPages() {
 	@Override
 	public boolean performFinish() {
 
-		try {
-			getContainer().run(true, true, new IRunnableWithProgress() {
+		// Extracting variable earlier to ensure consistency in asynchronous code flow
+		Set<IRepositoryModifiedObjects> pageOverwriteObjects = AbapGitWizardPull.this.pageOverwriteObjectsSelection.getSelectedObjects();
+		Set<IRepositoryModifiedObjects> pacgePackageWarningObjects = AbapGitWizardPull.this.pagePackageWarningObjectsSelection
+				.getSelectedObjects();
+
+		Job pullRepoJob = new Job("Pulling Repository") { //$NON-NLS-1$
+			@Override
+			protected IStatus run(IProgressMonitor monitor) {
+				try {
+					monitor.beginTask(Messages.AbapGitWizard_task_pulling_repository, 100);
 
-				@Override
-				public void run(IProgressMonitor monitor) throws InvocationTargetException, InterruptedException {
-					monitor.beginTask(Messages.AbapGitWizard_task_pulling_repository, IProgressMonitor.UNKNOWN);
 					IRepositoryService repoService = RepositoryServiceFactory.createRepositoryService(AbapGitWizardPull.this.destination,
 							monitor);
 
-					//get the selected objects to be pulled
+					// Get the selected objects to be pulled
 					AbapGitWizardPull.this.repoToSelectedObjects = AbapGitUIServiceFactory.createAbapGitPullService()
-							.getSelectedObjectsToPullforRepo(AbapGitWizardPull.this.pageOverwriteObjectsSelection.getSelectedObjects(),
-									AbapGitWizardPull.this.pagePackageWarningObjectsSelection.getSelectedObjects());
+							.getSelectedObjectsToPullforRepo(pageOverwriteObjects, pacgePackageWarningObjects);
 
-					//pull the selected objects
+					// Pull the selected objects
 					repoService.pullRepository(AbapGitWizardPull.this.selRepoData, AbapGitWizardPull.this.selRepoData.getBranchName(),
 							AbapGitWizardPull.this.transportPage.getTransportRequestNumber(), AbapGitWizardPull.this.cloneData.user,
 							AbapGitWizardPull.this.cloneData.pass,
 							AbapGitWizardPull.this.repoToSelectedObjects.get(AbapGitWizardPull.this.selRepoData.getUrl()), monitor);
 
+					// Pull dependencies if any
 					if (AbapGitWizardPull.this.cloneData.hasDependencies()) {
 						pullDependencies(monitor, repoService);
 					}
+
+					return Status.OK_STATUS;
+
```

Comment:
Can we refresh the repositories view after the job is finished ?
I am not very sure.